contracts/CreatorNFT.sol

```diff
@@ -0,0 +1,190 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.7;
+
+import "@openzeppelin/contracts/token/ERC721/extensions/ERC721URIStorage.sol";
+import "@openzeppelin/contracts/utils/Strings.sol";
+
+// checking ownership
+error CreatorNFT__InvalidCreator();
+// checking if the mint amount is > 0
+error CreatorNFT__InvalidMintAmount();
+// checking if the mint amount match the number of token uri
+error CreatorNFT__InvalidParams();
+// checking if the price is >= 0;
+error CreatorNFT__InvalidPrice();
+
+/**
+ * @title Client's own NFT contract
+ * @notice This contract provides functionalities for users to mint and set
+ *         royality fee when they create NFTs using NichoNFT platform
+ */
+contract CreatorNFT is ERC721URIStorage {
+    // helper function
+    using Strings for uint256;
+
+    // This event can be catched by the front-end to keep track of any activities
+    event Minted(uint indexed tokenId, string indexed tokenUri);
+
+    // assign an ownership
+    address private owner;
+    // keep track of token id
+    uint private tokenCounter;
+    // keep track of user commission in percentage
+    uint private commissionPercentage;
+    //keep track of each NFT price
+    mapping(uint => uint) private itemPrice;
+
+    // create an instance of ERC721 and ownership
+    constructor(address _owner) ERC721("CLIENT NFT", "CNFT") {
+        owner = _owner;
+    }
+
+    // checking if the amount is valid
+    modifier validAmount(uint _amount) {
+        if (_amount <= 0) revert CreatorNFT__InvalidMintAmount();
+        _;
+    }
+
+    // checking if the price is valid
+    modifier validPrice(uint _price) {
+        if (_price <= 0) revert CreatorNFT__InvalidPrice();
+        _;
+    }
+
+    /**
+     * @notice This function will allow creator to set their NFTs with a single price
+     *         instead of set the price 1 by 1 manually
+     * @param _price -> the price creator want to set
+     */
+    function setAllNftPrice(uint _price) external validPrice(_price) {
```

Comment:
Some external functions are exposed to any users. so you need to restrict to only owner to set the configurations.
I think it would be better to create modifier.